library/wizard/src/modules/Wizard.rb

```diff
@@ -1881,6 +1851,24 @@ def SetProductName(name)
     publish function: :AddMenuEntry, type: "list <map> (list <map>, string, string, string)"
     publish function: :CreateMenu, type: "void (list <map>)"
     publish function: :SetProductName, type: "void (string)"
+
+    protected
+
+    # Sets the icon and opens a wizard dialog with the content specified as
+    # arguments
+    def open_wizard_dialog(*args)
+      set_icon
+      UI.OpenDialog(
+        Opt(:wizardDialog),
+        Wizard(*args)
+      )
+    end
+
+    # Sets the application icon according to the value of @icon_name
```

Comment:
"This should be called only before an actual dialog is opened; premature UI calls can interfere with the CommandLine mode"